### PR TITLE
[wellconnstatus] New module well connection status.

### LINKED ISSUE #375: [wellconnstatus] Make new model for well connection status

### PR DESCRIPTION
**Pull request by lindjoha**
Fixes #375 

Reads CPI summary data and uses it to create a record of when each well connection is opened or closed. This functionality has been available in `webviz-subsurface

### CODE REVIEW

File: ecl2df/wellconnstatus.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+from ecl.summary import EclSum, EclSumKeyWordVector
+
+from ecl2df import getLogger_ecl2csv
+
+from .common import write_dframe_stdout_file
+
+
+def df(unsmry_file: str) -> pd.DataFrame:
+    """descr"""
+    eclsum = EclSum(unsmry_file, include_restart=False, lazy_load=False)
```
Comment (berland) [inline ecl2df/wellconnstatus.py:26]: Are you able to use `ecl2df.summary.df()` instead of punching libecl directly? That would help future maintenance.

Comment (lindjoha) [reply ecl2df/wellconnstatus.py:26]: Yes, I was wondering about that. I will try.

File: ecl2df/ecl2csv.py [hunk 2]
```diff
@@ -209,6 +209,14 @@ def get_parser() -> argparse.ArgumentParser:
         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
         description=("Each row represents an edge in the GRUPTREE at a specific date."),
     )
+    subparsers_dict["wellconnstatus"] = subparsers.add_parser(
+        "wellconnstatus",
+        help="Extract well connection status",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        description=(
+            "Each row represent an event where a well connection is changing status."
```
Comment (berland) [inline ecl2df/ecl2csv.py:217]: represent**s**

File: ecl2df/wellconnstatus.py (hunk 1 above)
Comment (lindjoha) [reply ecl2df/wellconnstatus.py]: Now I used `smry = summary.df(eclfiles, column_keys="CPI*")` and it made the code a lot simpler.

File: ecl2df/wellconnstatus.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+def _extract_single_connection_status_changes(
+    dates: np.ndarray, conn_values: np.ndarray
+) -> List[Tuple[Any, str]]:
+    """Extracts the status history of a single connection as a list of tuples
+    on the form (date, status)
+    """
+    status_changes = []
+    prev_value = 0
+    for date, value in zip(dates, conn_values):
+        if value > 0 and prev_value == 0:
```
[the remark this answers is not in the document]
Comment (lindjoha) [reply ecl2df/wellconnstatus.py:77]: Yes, I can write something about it in the docstring

**Comment by lindjoha** [comment]
> `@pytest.mark.parametrize` on unit tests, with a selection of inputs and outputs, and this is shaping up!

Yes, I am working on that!

File: ecl2df/wellconnstatus.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
+"""Exctracts connection status history for each compdat connection that
+is included in the summary data on the form CPI:WELL,I,J,K. One line is
+added to the export every time a connection changes status. It is OPEN when
+CPI>0 and SHUT when CPI=0. The earliest date for any connection will be OPEN,
+i.e a cell can not be SHUT before it has been OPEN. This means that any cells
+that are always SHUT will be excluded.
+
```
Comment (berland) [inline ecl2df/wellconnstatus.py:7]: some mention perhaps that "CPI" is "Connection Productivity Index" ?

**Comment by berland** [comment]
This module warrants a new entry in `docs/usage.rst` and a new file in `docs/usage/wellconnstatus`

**Pull request merged**